tensorflow_datasets/core/as_dataframe.py

```diff
@@ -140,14 +140,39 @@ def _repr_html_(self) -> str:
     return self.__styler._repr_html_()  # pylint: disable=protected-access
 
 
+def _flatten_with_path(structure: TreeDict):
```

Comment:
Could this be moved into `py_utils` ? And the tests in `py_utils_test`. Note: pytest support both unittest-style and pytest-style tests, so even if `py_utils_test.py` currently uses `unittest-style`, you can keep the pytest-style function which are better.